spec/ssh_spec.rb

```diff
@@ -120,7 +134,70 @@
       ).to receive(
         :error,
       ).with(
-        /mock_ssh_command.*mock_jump_user@mock_jump_host.*rossi@mock_host/,
+        /SSH returned error while connecting to #{tt_ssh.ssh_target}.*mock_ssh_command.*mock_jump_user@mock_jump_host.*rossi@mock_host/m,
+      )
+      expect { tt_ssh.error! }.to raise_error(
+        TasteTester::Exceptions::SshError,
+      )
+    end
+    after do
+      TasteTester::Config.restore(config_hash)
+    end
+  end
+
+  context 'test custom ssh configs linux - generated ssh command' do
+
+    let(:mock_generated_cmd) do
+      'mock_ssh_command_bin -o option1 -o option2 mock_user@mock_host'
+    end
+
+    before do
+      TasteTester::Config.ssh_connect_timeout 10
+      TasteTester::Config.jumps 'mock_jump_user@mock_jump_host'
+      TasteTester::Config.ssh_command 'mock_ssh_command'
+      TasteTester::Config.user 'rossi'
+      TasteTester::Config.ssh_cmd_template "mock_generator_cmd %{arg1} %{arg2} %{jumps}%{host} --user %{user} --get-command"
```

Comment:
Won't `%{jumps}%{host}` make `-J jumphosttargethost` instead of `-J jumphost targethost` ?

Reply:
yaaah

Reply:
While I'm at it... doesn't make much sense to pass in ssh_command and ssh_cmd_template, right?

Reply:
We do need `ssh_cmd_template` to trigger the code.

Reply:
no you're misunderstanding. if you define `ssh_cmd_template`, then nothing uses `ssh_command`, so that one isn't needed.